File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.js
/** 경수의 수 중, 귤 종류의 수가 최솟값인 경우 구하기 */
// 귤의 크기만큼 배열 생성 => 각 인덱스에 귤 갯수 세기 => 배열을 오름차순으로 정렬 => 필요한 귤 갯수만큼 인덱스 이동
// - k: 팔고 싶은 귤 수
// - tangerine: 귤의 크기 배열
function solution(k, tangerine) {
    // 가장 큰 귤
    const sizeCount = new Array(Math.max(...tangerine)+1).fill(0);
    
    // 귤 인덱스에 맞춰 갯수 세기
    for(let i = 0; i < tangerine.length; i++) {
        const current = tangerine[i];
        sizeCount[current] += 1;
    }
    sizeCount.sort((a, b) => b-a);
    
    // k만큼 고르기
    let index = 0;
    let answer = 0;
    while (k > 0) {
        k -= sizeCount[index];
        answer++; 
        index++;
    }
    
    // 계산이요
    return answer;
}

File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
# Python 기초 익히기
def solution(k, tangerine):
    # 파이썬 리스트 만들기: [초기값] * 리스트 길이
    sizeCount = [0] * (max(tangerine) + 1)

    # 귤 인덱스에 맞춰 갯수 세기
    # 파이썬 for문: for i in range(리스트 길이):
    for i in range(len(tangerine)):
        current = tangerine[i]
        sizeCount[current] += 1

    # 파이썬 내림차순 정렬: sort(reverse=True), 기본값 reverse=False
    # 특이한 점 > 메소드가 두 가지
    # - sort(): sort in place
    # - sorted(): returns sorted array, 원본 리스트는 그대로
    sizeCount.sort(reverse=True) 

    # k만큼 고르기
    index = 0
    answer = 0
    while k > 0:
        k -= sizeCount[index]
        answer += 1
        index += 1

    return answer